backend/app/controllers/api/v1/links_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Api::V1::LinksController < ApplicationController
+
+  before_filter :authenticate_user!, except: [:index]
```

Comment:
Add an after_action to verify authorized.
```
after_action :verify_authorized, except: [:index]
```

Also, use `_action` instead of `_filter`